### PR TITLE
Cleanup shotover::Runner api

### PR DESCRIPTION
**Pull request by rukai**
To give custom transform users a nicer setup process.

List of changes:
* User no longer has to provide ConfigOpts
* The user no longer has to manually set `with_observability_interface`, this method was originally added as a hack for specific integration tests but that hack is no longer needed so the function can just be removed and the observability interface always enabled.
* `Runner::new()` now internally performs the backup tracing reporter logic that was previously in `shotover-proxy/src/main.rs`
* run_block now returns `!`
* `Runner` renamed to `Shotover` as from the users perspective that type is shotover.

We could have just done a single `run_shotover()` method but `Shotover::new().run_block()` adds barely any more complication and gives us room to expand without breaking changes in the future.

### CODE REVIEW

File: shotover-proxy/src/runner.rs [hunk 1]
```diff
@@ -64,44 +65,74 @@ impl Default for ConfigOpts {
...
     runtime: Runtime,
     topology: Topology,
     config: Config,
     tracing: TracingState,
 }
 
-impl Runner {
-    pub fn new(params: ConfigOpts) -> Result<Self> {
+impl Shotover {
+    #[allow(clippy::new_without_default)]
```
Comment (conorbros) [inline shotover-proxy/src/runner.rs:76]: Why no default?

Comment (rukai) [reply shotover-proxy/src/runner.rs:76]: Its just a strange thing to want to do (just call new, its shorter!)
And I didnt want to have to commit to having default in our API.

File: shotover-proxy/src/runner.rs [hunk 2]
```diff
@@ -64,44 +65,74 @@ impl Default for ConfigOpts {
...
+        Ok(Shotover {
             runtime,
             topology,
             config,
             tracing,
         })
     }
 
-    pub fn with_observability_interface(self) -> Result<Self> {
+    pub fn start_observability_interface(
```
Comment (conorbros) [inline shotover-proxy/src/runner.rs:119]: This can be private?

**Review by conorbros — APPROVED** [review]
LGTM. Just that one comment

File: shotover-proxy/src/runner.rs (hunk 2 above)
Comment (rukai) [reply shotover-proxy/src/runner.rs]: good catch

**Review by benbromhead — COMMENTED** [review]
As this is a clean up for an external consumer of the api, do you want to add some doc strings to run_block?

**Review by benbromhead — APPROVED** [review]

**Pull request merged**